test/unit/core/store/middleware.ts

```diff
@@ -218,4 +261,78 @@ suite('Middleware', ({ expect, spy, stub }) => {
       expect(thunk).to.be.calledWithExactly(state);
     });
   });
+
+  describe('personalizationAnalyzer()', () => {
+    const conf = { a: 1 };
+    const state = { b: 2 };
+
+    it('should pass the action forward unchanged if not in list of relevant actions', () => {
+      const action = { a: 'b', type: 'NOT_VALID_ACTION' };
+      const config = stub(Selectors, 'config').returns(conf);
+      const enabled = stub(ConfigurationAdapter, 'isRealTimeBiasEnabled').returns(true);
+      const next = spy();
+      const getState = spy(() => state);
+
+      Middleware.personalizationAnalyzer(<any>{ getState })(next)(action);
+
+      expect(next).to.be.calledWithExactly(action);
+      expect(config).to.be.calledWithExactly(state);
+      expect(enabled).to.be.calledWithExactly(conf);
+      expect(getState).to.be.calledWithExactly();
+    });
+
+    it('should pass the action forward unchanged if real time biasing disabled', () => {
+      const action = { a: 'b', type: PERSONALIZATION_CHANGE_ACTIONS[0] };
+      const config = stub(Selectors, 'config').returns(conf);
+      const enabled = stub(ConfigurationAdapter, 'isRealTimeBiasEnabled').returns(false);
+      const next = spy();
+      const getState = spy(() => state);
+
+      Middleware.personalizationAnalyzer(<any>{ getState })(next)(action);
+
+      expect(next).to.be.calledWithExactly(action);
+      expect(config).to.be.calledWithExactly(state);
+      expect(enabled).to.be.calledWithExactly(conf);
+      expect(getState).to.be.calledWithExactly();
+    });
+
+    it('should pass the action forward if extractBias returns falsy', () => {
+      const action = { a: 'b', type: PERSONALIZATION_CHANGE_ACTIONS[0] };
+      const returnAction = 'return';
+      const extracted = 'extra';
+      const config = stub(Selectors, 'config').returns(conf);
+      const updateBiasing = stub(ActionCreators, 'updateBiasing').returns(returnAction);
+      const extract = stub(PersonalizationAdapter, 'extractBias').returns(null);
+      const next = spy();
+      const getState = spy(() => state);
+      stub(ConfigurationAdapter, 'isRealTimeBiasEnabled').returns(true);
+
+      Middleware.personalizationAnalyzer(<any>{ getState })(next)(action);
+
+      expect(next).to.be.calledWithExactly(action);
+      expect(extract).to.be.calledWithExactly(action, state);
+      expect(updateBiasing).to.not.be.called;
+      expect(config).to.be.calledWithExactly(state);
```

Comment:
we also don't need to duplicate all of these tests out for every `it()` as we've already covered them in other tests